Kernel/Syscalls/thread.cpp

```diff
@@ -100,24 +100,27 @@ void Process::sys$exit_thread(Userspace<void*> exit_value, Userspace<void*> stac
         crash(SIGABRT, {});
     }
 
-    if (this->thread_count() == 1) {
-        // If this is the last thread, instead kill the process.
-        this->sys$exit(0);
-    }
+    m_thread_count.with([&](auto& count) {
+        if (count == 1) {
+            // If this is the last thread, instead kill the process.
+            this->sys$exit(0);
+        }
 
-    auto* current_thread = Thread::current();
-    current_thread->set_profiling_suppressed();
-    PerformanceManager::add_thread_exit_event(*current_thread);
-
-    if (stack_location) {
-        auto unmap_result = address_space().with([&](auto& space) {
-            return space->unmap_mmap_range(stack_location.vaddr(), stack_size);
-        });
-        if (unmap_result.is_error())
-            dbgln("Failed to unmap thread stack, terminating thread anyway. Error code: {}", unmap_result.error());
-    }
+        auto* current_thread = Thread::current();
+        current_thread->set_profiling_suppressed();
+        PerformanceManager::add_thread_exit_event(*current_thread);
+
+        if (stack_location) {
+            auto unmap_result = address_space().with([&](auto& space) {
```

Comment:
Same here. why do we need to do all of this work with the thread count locked?

Reply:
If we lock the thread count, we ensure no new threads could be created, so I thought that this might be a good measure against such condition.